include/my_global.h

```diff
@@ -672,12 +672,17 @@ typedef SOCKET_SIZE_TYPE size_socket;
 #endif
 
 /*
-  Io buffer size; Must be a power of 2 and a multiple of 512. May be
-  smaller what the disk page size. This influences the speed of the
-  isam btree library. eg to big to slow.
+  I/O buffer size. May be smaller than the disk page size.
+  This influences the speed of the isam btree library. E.g.: too big too slow.
   4096 is a common block size on SSDs.
 */
+#ifndef IO_SIZE
 #define IO_SIZE			4096U
+#endif
+#if ((IO_SIZE <= 0) || ((IO_SIZE % 512) != 0) || ((IO_SIZE & (IO_SIZE-1)) != 0))
```

Comment:
This can be shortened to
```c
#if IO_SIZE < 512 || (IO_SIZE & (IO_SIZE - 1))
```

Reply:
Yes. And with the text in the `#error`, I agree that it remains clear-enough. I shall adjust it.